From 4eb35a5d37936a7097a16fb577f3789cdd8fb978 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=E9=BB=84=E6=B6=9B?= <huangtao@rock-chips.com>
Date: Tue, 28 Sep 2010 15:28:34 +0800
Subject: [PATCH] Revert "rk2818_fb: use codec pll"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This reverts commit d9479fe5d6b897cf980398e9537f8f67b8b495fa.

当以下条件同时满足时，RK281x的LCDC内部硬件存在bug，可能会导致概率性的横条纹现象。
（1）LCDC_DCLK不是ARM_PLL分频；
（2）win0视频进行旋转；
（3）win0视频格式为YUV4201（拼凑格式）；
（4）在视频宽度上（旋转后为高度）有缩小时；

解决方案：在硬件bug解除之前，在以上的应用场景下，保证LCDC_DCLK时钟源为ARM_PLL分频。
---
 drivers/video/rk2818_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rk2818_fb.c b/drivers/video/rk2818_fb.c
index 614c5a346255..41ab0a983baf 100755
--- a/drivers/video/rk2818_fb.c
+++ b/drivers/video/rk2818_fb.c
@@ -2307,7 +2307,7 @@ static int __init rk2818fb_probe (struct platform_device *pdev)
 		ret = -ENOENT;
 		goto unregister_win1fb;
 	}
-    inf->dclk_parent = clk_get(&pdev->dev, "codec_pll");
+    inf->dclk_parent = clk_get(&pdev->dev, "arm_pll");
     if (!inf->dclk_parent || IS_ERR(inf->dclk_parent))
     {
 		printk(KERN_ERR "failed to get lcd dclock parent source\n");
-- 
2.35.3

